src/pkg/p2p/preheat/models/policy/policy.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/astaxie/beego/validation"
+	"github.com/robfig/cron"
+
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/models/provider"
+)
+
+const (
+	// Filters:
+	//  Repository : type=Repository value=name text (double star pattern used)
+	//  Tag: type=Tag value=tag text (double star pattern used)
+	//  Signature: type=Signature value=bool (True/False)
+	//  Vulnerability: type=Vulnerability value=Severity (expected bar)
+	//  Label: type=Label value=label array
+	FilterTypeRepository    FilterType = "repository"
+	FilterTypeTag           FilterType = "tag"
+	FilterTypeSignature     FilterType = "signature"
+	FilterTypeVulnerability FilterType = "vulnerability"
+	FilterTypeLabel         FilterType = "label"
+
+	// trigger types
+	TriggerTypeManual     TriggerType = "manual"
+	TriggerTypeScheduled  TriggerType = "scheduled"
+	TriggerTypeEventBased TriggerType = "event_based"
+)
+
+// Policy defines p2p preheat policy
+type Policy struct {
```

Comment:
policy.Policy? Do we have a better name?

How about ` policy.Object` \ `policy.Definition` \ `policy.Schema`? / policy.?

Reply:
I think policy.Schema is good!